WelsonJS.Toolkit/WelsonJS.Launcher/ResourceServer.cs

```diff
@@ -133,44 +142,37 @@ private void ServeCompletion(HttpListenerContext context, string word)
                     ))
                 );
 
-                byte[] data = Encoding.UTF8.GetBytes(
-                    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" +
-                    response.ToString()
-                );
+                ServeResource(context, response.ToString(), "application/xml");
+            }
+            catch (Exception ex)
+            {
+                ServeResource(context, $"<error>Failed to process completion request. {ex.Message}</error>", "application/xml", 500);
+            }
+        }
 
-                context.Response.StatusCode = statusCode;
-                context.Response.ContentType = "application/xml";
-                context.Response.ContentLength64 = data.Length;
-                using (Stream outputStream = context.Response.OutputStream)
-                {
-                    outputStream.Write(data, 0, data.Length);
-                }
+        private void ServeDevTools(HttpListenerContext context, string endpoint)
+        {
+            try
+            {
+                HttpClient client = new HttpClient();
+                string url = "http://localhost:9222" + endpoint;
+                string data = Task.Run(async () => await client.GetStringAsync(url)).Result;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential SSRF (Server-Side Request Forgery) vulnerability detected. This pattern identifies code structures that might allow attackers to manipulate server-side requests, posing a risk of unauthorized access and data leaks. Ensure that user-controlled inputs in HttpClient, WebClient, or RestClient requests are thoroughly validated and sanitized to mitigate this security threat.